pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -689,41 +684,17 @@ func getSplunkCredential(client client.Client) (*render.SplunkCredential, error)
 		if errors.IsNotFound(err) {
 			return nil, nil
 		}
-		return nil, fmt.Errorf("Failed to read secret %q: %s", render.SplunkFluentdTokenSecretName, err)
+		return nil, fmt.Errorf("failed to read secret %q: %s", render.SplunkFluentdTokenSecretName, err)
 	}
 
-	var ok bool
-	var token []byte
-	if token, ok = tokenSecret.Data[render.SplunkFluentdSecretTokenKey]; !ok || len(token) == 0 {
-		return nil, fmt.Errorf(
-			"Expected secret %q to have a field named %q",
+	token, ok := tokenSecret.Data[render.SplunkFluentdSecretTokenKey]
+	if !ok || len(token) == 0 {
+		return nil, fmt.Errorf("expected secret %q to have a field named %q",
 			render.SplunkFluentdTokenSecretName, render.SplunkFluentdSecretTokenKey)
 	}
 
-	var certificate []byte
-	certificateSecret := &corev1.Secret{}
-	certificateNamespacedName := types.NamespacedName{
-		Name:      render.SplunkFluentdCertificateSecretName,
-		Namespace: common.OperatorNamespace(),
-	}
-
-	if err := client.Get(context.Background(), certificateNamespacedName, certificateSecret); err != nil {
-		if errors.IsNotFound(err) {
-			log.Info(fmt.Sprintf("Splunk certificate secret %v not provided. Assuming http protocol or trusted CA certificate.",
-				render.SplunkFluentdCertificateSecretName))
-		} else {
-			return nil, fmt.Errorf("Failed to read secret %q: %s", render.SplunkFluentdCertificateSecretName, err)
-		}
-	} else {
-		if certificate, ok = certificateSecret.Data[render.SplunkFluentdSecretCertificateKey]; !ok || len(certificate) == 0 {
```

Comment:
Can't we just add it to the trusted bundle here?

Reply:
Yeah. We could. As this isn't really a customer request before, I would like to remove it until we get a real requirement.